contracts/generic/AugmintToken.sol

```diff
@@ -30,12 +33,39 @@ contract AugmintToken is AugmintTokenInterface {
         feeAccount = _feeAccount;
 
     }
-
     function transfer(address to, uint256 amount) external returns (bool) {
         _transfer(msg.sender, to, amount, "");
         return true;
     }
 
+    /* Transfers based on an offline signed transfer instruction. */
+    function delegatedTransfer(address from, address to, uint amount, string narrative,
+                                     uint minGasPrice,  /* client provided gasPrice on which she expects tx to be exec. */
+                                     uint maxExecutorFee, /* client provided max fee for executing the tx */
+                                     bytes32 nonce, /* random nonce generated by client */
+                                     /* ^^^^ end of signed data ^^^^ */
+                                     bytes signature,
+                                     uint requestedExecutorFee /* the executor can decide to request lower fee */
+                                     )
+    external returns(bool) {
+
+        require(!noncesUsed[nonce], "nonce already used");
```

Comment:
the nonce should be included in a hash along with all other parameters (or just use the signature) and we should store the used hashes (not the nonce). otherwise there's a security issue: I could censor transactions by front-running the nonce from the mem-pool.

Reply:
also, you could get away with much less bits for the nonce then. you just need to avoid collisions between transactions with the same from/to/amount.

Reply:
1. nonce was already included in the hash
2. pushing a change to store hashes used instead of nonce
3. lower nonce bits:  based on my tests passing+keccak256 of uint32 or bytes4 types actually consume slightly  more gas than uint256 or bytes32. Maybe keccak256 is padding each param to same size?